xbmc/interfaces/json-rpc/SystemOperations.cpp

```diff
@@ -95,6 +96,21 @@ JSONRPC_STATUS CSystemOperations::Reboot(const std::string &method, ITransportLa
     return FailedToExecute;
 }
 
+JSONRPC_STATUS CSystemOperations::InhibitIdleShutdown(const std::string &method, ITransportLayer *transport, IClient *client, const CVariant &parameterObject, CVariant &result)
+{
+  if ((parameterObject["inhibit"].isString() &&
+       parameterObject["inhibit"].asString().compare("toggle") == 0) ||
+      (parameterObject["inhibit"].isBoolean() &&
+       parameterObject["inhibit"].asBoolean() != g_application.IsIdleShutdownInhibited()))
+  {
+    CApplicationMessenger::GetInstance().SendMsg(TMSG_INHIBITIDLESHUTDOWN, !g_application.IsIdleShutdownInhibited());
+  }
+  else if (!parameterObject["inhibit"].isBoolean() && !parameterObject["inhibit"].isString())
```

Comment:
This can't actually happen because the JSON-RPC system makes sure that the parameters match the API definition.